pkg/gatherers/workloads/gather_workloads_runtime_infos.go

```diff
@@ -0,0 +1,215 @@
+					}
+				}
+			}
+		}
+	}
+}
+
+// Transform the workload data from the insights operator runtime component (represented by
+// a map of namespace/pod-name/container-id/insightsWorkloadRuntimeInfo) into the data
+// stored by the Insights operator as a map of namespace/pod-name/container-id/workloadRuntimeInfoContainer
+// where each value is hashed.
+func transformWorkload(h hash.Hash,
+	node map[string]map[string]map[string]insightsWorkloadRuntimeInfo,
+) workloadRuntimes {
+
+	result := make(workloadRuntimes)
+
+	for podNamespace, podWorkloads := range node {
+		result[podNamespace] = make(map[string]map[string]workloadRuntimeInfoContainer)
+		for podName, containerWorkloads := range podWorkloads {
+			result[podNamespace][podName] = make(map[string]workloadRuntimeInfoContainer)
+			for containerID, info := range containerWorkloads {
+				runtimeInfo := workloadRuntimeInfoContainer{
+					Os:              hashString(h, info.OSReleaseID),
+					OsVersion:       hashString(h, info.OSReleaseVersionID),
+					Kind:            hashString(h, info.RuntimeKind),
+					KindVersion:     hashString(h, info.RuntimeKindVersion),
+					KindImplementer: hashString(h, info.RuntimeKindImplementer),
+				}
+
+				runtimeInfos := make([]RuntimeComponent, len(info.Runtimes))
+				for i, runtime := range info.Runtimes {
+					runtimeInfos[i] = RuntimeComponent{
+						Name:    hashString(h, runtime.Name),
+						Version: hashString(h, runtime.Version),
+					}
+				}
+				runtimeInfo.Runtimes = runtimeInfos
+				result[podNamespace][podName][containerID] = runtimeInfo
+
+			}
+		}
+	}
+	return result
+}
+
+// hashString return a hash of the string if it is not empty (or the empty string otherwise)
+func hashString(h hash.Hash, s string) string {
+	if s == "" {
+		return s
+	}
+	return workloadHashString(h, s)
+}
+
+// Get all WorkloadRuntimeInfos for a single Node (using the insights-operator-runtime pod running on this node)
+func getNodeWorkloadRuntimeInfos(
+	ctx context.Context,
+	h hash.Hash,
+	coreClient corev1client.CoreV1Interface,
+	restConfig *rest.Config,
+	runtimePodName string,
+) workloadRuntimes {
+	execCommand := []string{"/scan-containers"}
+
+	req := coreClient.RESTClient().
+		Post().
+		Namespace(os.Getenv("POD_NAMESPACE")).
+		Name(runtimePodName).
+		Resource("pods").
+		SubResource("exec").
+		VersionedParams(&corev1.PodExecOptions{
+			Command: execCommand,
+			Stdout:  true,
+			Stderr:  true,
+		}, scheme.ParameterCodec)
+
+	exec, err := remotecommand.NewSPDYExecutor(restConfig, "POST", req.URL())
+	if err != nil {
+		fmt.Printf("error: %s", err)
+	}
+	var (
+		execOut bytes.Buffer
+		execErr bytes.Buffer
+	)
+
+	err = exec.StreamWithContext(ctx, remotecommand.StreamOptions{
+		Stdout: &execOut,
+		Stderr: &execErr,
+		Tty:    false,
+	})
+	if err != nil {
+		fmt.Printf("got insights operator runime error: %s\n", err)
+		fmt.Printf("command error output: %s\n", execErr.String())
+		fmt.Printf("command output: %s\n", execOut.String())
```

Comment:
I guess this all is TBD, but the errors should be logged properly or returned by the function (and handled later on)